eng/tools/dependency-testing/index.js

```diff
@@ -95,15 +95,21 @@ async function insertPackageJson(
   testFolder,
 ) {
   const testPath = path.join(targetPackagePath, testFolder);
-  let templateJson = await packageUtils.readFileJson("./templates/package.json");
-  let testPackageJson = templateJson;
+  const testPackageJson = await packageUtils.readFileJson("./templates/package.json");
   if (packageJsonContents.name.startsWith("@azure/")) {
     testPackageJson.name = packageJsonContents.name.replace("@azure/", "azure-") + "-test";
   } else if (packageJsonContents.name.startsWith("@azure-rest/")) {
     testPackageJson.name =
       packageJsonContents.name.replace("@azure-rest/", "azure-rest-") + "-test";
   }
   await usePackageTestTimeout(testPackageJson, packageJsonContents);
+  testPackageJson.type = packageJsonContents.type;
+  if (packageJsonContents.scripts["integration-test:node"].includes("vitest")) {
+    testPackageJson.scripts["integration-test:node"] = "dev-tool run test:vitest -- -c vitest.dependency-test.config.ts";
```

Comment:
why not running with the library's original vitest config? this is causing issues for event hubs tests

Reply:
For dependency testing we essentially create a new package under test/public.  We could point to the original vitest config if the paths etc. still work.